contracts/Staking.sol

```diff
@@ -74,18 +77,22 @@ contract Staking is Ownable {
     @param tokenAddress Token address
     @param data should be disregarded for the current deployment
      */
-    function stakeFor(uint256 amount, address user, address tokenAddress, bytes memory data) public payable {
-        require(isInWhiteList(tokenAddress), "Staking: not possible to interact with this token");
+    function stakeFor(
+        uint256 amount,
+        address user,
+        address tokenAddress,
+        bytes memory data
+    ) public payable whitelistedToken(tokenAddress) {
         // disregard passed-in amount
-        if(_isNativeToken(tokenAddress)) {
+        if (_isNativeToken(tokenAddress)) {
```

Comment:
Maybe it makes sense to create a `memory` variable?
```
bool memory isNativeToken = _isNativeToken(tokenAddress)
```
and then use it in both checks. Might save us some gas.

Reply:
Make sense.